src/sentry/replays/tasks.py

```diff
@@ -20,16 +22,22 @@
 )
 def delete_recording_segments(project_id: int, replay_id: str, **kwargs: dict) -> None:
     """Asynchronously delete a replay."""
-    archive_replay(project_id, replay_id)
     delete_replay_recording(project_id, replay_id)
+    archive_replay(project_id, replay_id)
 
 
 def delete_replay_recording(project_id: int, replay_id: str) -> None:
     """Delete all recording-segments associated with a Replay."""
-    segments = ReplayRecordingSegment.objects.filter(
+    segments = fetch_segments_metadata(project_id, replay_id, offset=0, limit=10000)
+    for segment in segments:
+        driver = FilestoreBlob() if segment.file_id else StorageBlob()
+        driver.delete(segment)
+
+    # delete old rows from SQL too
+    segment_sql_rows = ReplayRecordingSegment.objects.filter(
```

Comment:
Does it make sense to remove the rows first in the event process dies before blob deletion is supposed to run?
Concern is: If this fails midway we'll have rows which will create a failure mode elsewhere in the product (links render, failing to fetch blob?).

Reply:
these rows refer to the FileStore metadata blobs. the above loop deletes the blobs -- if either failed the outcome would still be the same. the clickhouse rows are deleted in `archive_replay`. right now we aren't deleting the blobs at all for GCS, so this is already the case in the product.